File: agent.js
// agent.js
const http = require('http');
const fs = require('fs');
const path = require('path');

const server = http.createServer((req, res) => {
    if (req.method === 'POST') {
        let body = '';
        req.on('data', chunk => body += chunk);
        req.on('end', () => {
            try {
                const data = JSON.parse(body);
                const dir = data.target_dir || path.join(require('os').homedir(), '云枢记忆');
                fs.mkdirSync(dir, { recursive: true });
                const file = path.join(dir, data.filename || '云枢记忆.txt');
                fs.appendFileSync(file, data.content + '\n\n');
                res.end(JSON.stringify({ status:'ok' }));
            } catch (e) {
                res.end(JSON.stringify({ status:'error', message: e.message }));
            }
        });
    } else {
        res.statusCode = 405;
        res.end('Method Not Allowed');
    }
});

server.listen(8123, '127.0.0.1', () => {
    console.log('云枢Agent启动，监听端口8123');
});

File: yunshu/agent.py
```python
import http.server
import json
import os
import urllib.parse

class AgentHandler(http.server.BaseHTTPRequestHandler):
    def do_POST(self):
        length = int(self.headers['Content-Length'])
        data = json.loads(self.rfile.read(length).decode('utf-8'))
        action = data.get('action', '')
        target_dir = data.get('target_dir', os.path.expanduser('~/云枢记忆'))
        os.makedirs(target_dir, exist_ok=True)
        
        if action == 'save':
            filename = data.get('filename', '云枢记忆.txt')
            content = data.get('content', '')
            filepath = os.path.join(target_dir, filename)
            with open(filepath, 'a', encoding='utf-8') as f:
                f.write(content + '\n\n')
            self.send_response(200)
            self.end_headers()
            self.wfile.write(json.dumps({"status":"ok"}).encode())
        else:
            self.send_response(400)
            self.end_headers()

    def log_message(self, format, *args):
        pass  # 不输出日志

if __name__ == '__main__':
    server = http.server.HTTPServer(('127.0.0.1', 8123), AgentHandler)
    print('云枢Agent启动，监听端口8123')
    server.serve_forever()
```